Add support for LTM4673

cencarna opened the pull request:
## PR Description

This adds support for LTM4673 to the LTC2978 PMBus client driver which contains
support to similar μModule regulators. The LTM4673 is a quad output, dual 12A and
dual 5A, switching mode DC/DC step-down μModule regulator.

## PR Type
- [ ] Bug fix (a change that fixes an issue)
- [x] New feature (a change that adds new functionality)
- [ ] Breaking change (a change that affects other repos or cause CIs to fail)

## PR Checklist
- [x] I have conducted a self-review of my own code changes
- [x] I have tested the changes on the relevant hardware
- [ ] I have updated the documentation outside this repo accordingly (if there is the case)

cencarna committed:
hwmon: (pmbus/ltc2978) add support for ltm4673

Add support for LTM4673. The LTM4673 is a quad output, dual 12A and dual
5A, switching mode DC/DC step-down μModule regulator integrated with
4-channel power system manager.

This adds only the chip id, the checks for the manufacturer special id,
and the relevant attributes for the device's pmbus_driver_info.
The device does not support clear peaks.

Signed-off-by: Cedric Encarnacion <cedricjustine.encarnacion@analog.com>
Link: https://lore.kernel.org/r/20250124-ltm4673-v1-2-a2c6aa37c903@analog.com
Signed-off-by: Guenter Roeck <linux@roeck-us.net>

nunojsa commented on Documentation/hwmon/ltc2978.rst:
```diff
@@ -227,6 +235,7 @@ Description
 - LTM2987 is a 16-channel Power System Manager with two LTC2977 plus
 - additional components on a single die. The chip is instantiated and reported
 - as two separate chips on two different I2C bus addresses.
+- LTM4673 is a dual 12A and dual 5A μModule regulator.
```
Not likely a problem but I think you squash this change with the patch adding the driver.

nunojsa reviewed (commented):
Seems good to me. Please go ahead and send it upstream

nunojsa commented:
@cencarna, Is this sent?

cencarna committed:
dt-bindings: hwmon: ltc2978: add support for ltm4673

Add LTM4673 to supported devices of LTC2978. Unlike other LTM46xx
devices, LTM4673 is a μModule regulator that outputs four channels.

Signed-off-by: Cedric Encarnacion <cedricjustine.encarnacion@analog.com>
Acked-by: Krzysztof Kozlowski <krzysztof.kozlowski@linaro.org>
Link: https://lore.kernel.org/r/20250124-ltm4673-v1-1-a2c6aa37c903@analog.com
Signed-off-by: Guenter Roeck <linux@roeck-us.net>
